nemo_curator/scripts/semdedup/compute_embeddings.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import time
+from typing import List
+
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.log import create_logger
+from nemo_curator.modules.config import SemDedupConfig
+from nemo_curator.modules.semantic_dedup import EmbeddingCreator
+from nemo_curator.utils.distributed_utils import get_client, read_data
+from nemo_curator.utils.file_utils import get_remaining_files
+from nemo_curator.utils.script_utils import ArgumentHelper
+
+
+def get_input_files(
+    input_data_dir: str, input_file_type: str, output_data_dir: str, num_samples: int
+) -> List[str]:
+    os.makedirs(output_data_dir, exist_ok=True)
+    len_written_files = len(os.listdir(output_data_dir))
+    input_files = get_remaining_files(input_data_dir, output_data_dir, input_file_type)
+    # Gaurd against non-json files present in the input directory
+    input_files = [f for f in input_files if f.endswith(input_file_type)]
+    if num_samples > 0:
+        if len_written_files > num_samples:
+            left_to_sample = 0
+        else:
+            left_to_sample = num_samples - len_written_files
+    else:
+        left_to_sample = len(input_files)
+
+    input_files = input_files[:left_to_sample]
```

Comment:
We could also consider merging in this file limit code to `get_remaining_files` as well if you think it would be generally useful. Though I do want to ask, why in particular is this file limit code useful in the case of the embedding calculation?

Reply:
The reason is everything downstream of embedding creation just works on top of embeddings and this is an order of magnitude more expensive operation .

Reply:
True, but is there any particular reason why a user would only want to consider the first n files instead of all of them (other than for the sake of experimentation). I'm fine keeping this as it is, I'm just curious.